consensus/chaos/chaos.go

```diff
@@ -1019,6 +1019,10 @@ func encodeSigHeader(w io.Writer, header *types.Header) {
 }
 
 func (c *Chaos) PreHandle(chain consensus.ChainHeaderReader, header *types.Header, state *state.StateDB) error {
+	if c.chainConfig.Hardfork1Block != nil && c.chainConfig.Hardfork1Block.Uint64() > 0 &&
+		c.chainConfig.Hardfork1Block.Cmp(header.Number) == 0 {
```

Comment:
I think `c.chainConfig.Hardfork1Block.Cmp(header.Number) == 0` contains the condition `c.chainConfig.Hardfork1Block.Uint64() > 0`. Maybe `c.chainConfig.Hardfork1Block.Cmp(header.Number) == 0` alone is suffient.